streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorNodeTest.java

```diff
@@ -63,6 +95,36 @@ public void shouldThrowStreamsExceptionIfExceptionCaughtDuringClose() {
         assertThrows(StreamsException.class, () -> node.init(null));
     }
 
+    @Test
+    public void shouldThrowStreamsExceptionWhenProcessingMarkedAsFail() {
+        final ProcessorNode<Object, Object, Object, Object> node =
+            new ProcessorNode<>("name", new ProcessingExceptionProcessor(), Collections.emptySet());
+        node.setProcessingExceptionHandler(new LogAndFailProcessingExceptionHandler());
```

Comment:
I am fine if you want to use `ProcessingExceptionHandlerMockTest` but could you please rename it to `ProcessingExceptionHandlerMock` so that it is clear that it is a mock and not a test. 
Could you also use constants in `mockInternalProcessorContext()` for the returning values of the stubs and correspondingly use the same constants in the assertions in `handle()` of `ProcessingExceptionHandlerMock`. It is not important that `context.topic()` returns `"topic"`, it is important that it returns the same value as set in the processor context.

Reply:
I added one test to compare results in the ProcessingExceptionHandlerMock (I removed the Test)

Reply:
@cadonna Fixed, the comparison is now done against the internal processor context:

```java
assertEquals(internalProcessorContext.topic(), context.topic());
```

When creating `mockInternalProcessorContext()`, should the returned values be explicitely declared `final`?

Reply:
That looks great!

Reply:
> When creating mockInternalProcessorContext(), should the returned values be explicitely declared final?

I am sorry, but I do not understand the question. What do you mean with "returned values"?

Reply:
I was only wondering if this would be more elegant:

```java
public class ProcessorNodeTest {
    private final String topic = "topic";
    private final int partition = 0;
    private final Long offset = 0L;

    private InternalProcessorContext<Object, Object> mockInternalProcessorContext() {
        final InternalProcessorContext<Object, Object> internalProcessorContext = mock(InternalProcessorContext.class);

        when(internalProcessorContext.topic()).thenReturn(topic);
        when(internalProcessorContext.partition()).thenReturn(partition);
        when(internalProcessorContext.offset()).thenReturn(offset);

        return internalProcessorContext;
    }
}
```

It does not matter that much

Reply:
@loicgreffier 
I would even do:
```java
public class ProcessorNodeTest {
    private static final String TOPIC = "topic";
    private static final int PARTITION = 0;
    private static final Long OFFSET = 0L;

    private InternalProcessorContext<Object, Object> mockInternalProcessorContext() {
        final InternalProcessorContext<Object, Object> internalProcessorContext = mock(InternalProcessorContext.class);

        when(internalProcessorContext.topic()).thenReturn(TOPIC);
        when(internalProcessorContext.partition()).thenReturn(PARTITION);
        when(internalProcessorContext.offset()).thenReturn(OFFSET);

        return internalProcessorContext;
    }
}
```
and use those constants in all test methods.

Let me know, if you want to make these changes, otherwise I am ready to merge this PR.

Reply:
Not sure it is really necessary, as those contants are only used in the mockInternalProcessorContext() method.

If you agree with that. You can merge this PR.  :-)

thanks

Reply:
It is not necessary, but it is cleaner. BTW, some constants are used outside of `mockInternalProcessorContext()`, for example `new TaskId(0, 0)`. But even those that are only used in `mockInternalProcessorContext()` are used there multiple times, for example offset, topic, and partition.

Reply:
@cadonna 

I can add thoses constants and use them in all methods: 

```java
    private static final String TOPIC = "topic";
    private static final int PARTITION = 0;
    private static final Long OFFSET = 0L;
    private static final Long TIMESTAMP = 0L;
    private static final TaskId TASK_ID = new TaskId(0, 0);
    private static final String NAME = "name";
    private static final String KEY = "key";
    private static final String VALUE = "value";
```
  
is it ok for you ?

Reply:
Looks good!